enhancements/ingress/aws-lb-eip-selection.md

```diff
@@ -0,0 +1,474 @@
+
+Throw a simple error that just says, EIP != Provided Subnets:
+The number of EIP Allocations does not equal the number of provided Subnets, the cluster will fail.
+
+##### BYO Subnet Count == EIPs Allocations && BYO Subnet Count != Predicted LB Subnet count && Predicted LB Subnet count != EIPs Allocations:
+
+Throw a error that says something to the extend of that the LB will select more subnets and the cluster will fail:
+Or though the number of EIP Allocations equals the number of provided Subnets, it does not equal the number of Subnets that will be selected by
+the LoadBalancer and the cluster will fail, please review the kubernetes.io/cluster/<cluster-id> subnet tags are set correctly.
+
+##### BYO Subnet Count != EIPs Allocations && BYO Subnet Count != Predicted LB Subnet count && Predicted LB Subnet count == EIPs Allocations:
+
+This is an odd scenario. The user got the # of a EIPs == Predicted LB Subnet count, I suppose because they anticipated the AWS CCM's generous 
+selection of subnets. This is valid scenario for no error message.
+
+##### BYO Subnet Count == EIPs Allocations && BYO Subnet Count == Predicted LB Subnet count && Predicted LB Subnet count == EIPs Allocations:
+
+Obvious valid scenario.
+
+I think we need some feedback from installer-team, @patrickdillon or @mtulio or @sadasu on this type of validation. Should the installer team consider
+the Predicted LB Subnets != BYO Subnet Count scenario as not valid? And possibly block future installs as a resolution
+to https://issues.redhat.com/browse/OCPBUGS-17432? That would make EIP Allocation a lot easier, but not sure if that's realistic. 
+
+4. #### Validation to check if EIPs exist and are not already assigned to resources.
+EIPs can be assigned to many resource types, like Nat Gateways, *LBs, Instances, etc. The attribute associationId will be set when the EIP is already associated.
+To mitigate this we could add that validation, at least on installer, to provide quick-feedback (fail when validate install-config) to the user when the provided EIP is already associated to another resource.
+It would be nice to have a validation before setting the annotation to CCM, keeping the operator degraded before disrupting the service.
+
+5. #### IngressConfig EIP Allocation Defaulting Mechanics for Ingress Controller
+Traditionally, the Ingress Operator has populated default values from the Ingress Config into the `status`, 
+making `status` effectively reflect the desired state of the IngressController. However, since 
+`eipAllocations` in `status` represents the **actual** state, not the **desired** state, the default 
+`eipAllocations` values must be set in the `spec` when the Ingress Operator initially admits the 
+IngressController.
+This approach is new. The Ingress Operator does not typically set default values in `spec` for load balancer
+configurations if the user hasn’t explicitly provided them. While this defaulting pattern is more consistent
```

Comment:
The admin user has explicitly provided them, as a part of the installation config.